components/suggest/src/db.rs

```diff
@@ -216,12 +223,60 @@ impl<'a> SuggestDao<'a> {
                                         number_of_ratings: row.get("number_of_ratings")?,
                                         guid: row.get("guid")?,
                                         score: row.get("score")?,
-                                    })
+                                    }))
                             })
                     },
+                    SuggestionProvider::Pocket => {
+                        let confidence = row.get("confidence")?;
+                        self.conn.query_row_and_then(
+                            "SELECT keyword_suffix FROM pocket_keywords k 
+                             WHERE k.suggestion_id = :suggestion_id 
+                             AND k.keyword_prefix = :keyword_prefix", 
+                            named_params! {
+                                ":suggestion_id": suggestion_id, 
+                                ":keyword_prefix": keyword_prefix 
+                            },
+                            |row| {
+                                let suffixes_match = match confidence {
+                                    KeywordConfidence::Low => row.get::<_, String>("keyword_suffix")?.starts_with(keyword_suffix),
+                                    KeywordConfidence::High => row.get::<_, String>("keyword_suffix")? == keyword_suffix,
+                                };
+                                if suffixes_match {
+                                    self.conn.query_row_and_then(
+                                        "SELECT p.score
+                                        FROM pocket_custom_details p
+                                        WHERE p.suggestion_id = :suggestion_id",
+                                        named_params! {
+                                            ":suggestion_id": suggestion_id
+                                        },
+                                        |row| {
+                                            Ok(Some(Suggestion::Pocket {
+                                                title,
+                                                url: raw_url,
+                                                score: row.get("score")?,
+                                                is_top_pick: matches!(
+                                                    confidence,
+                                                    KeywordConfidence::High
+                                                )
+
+                                            }))
+                                        }
+                                    )
+                                }
+                                else{
```

Comment:
```suggestion
                                } else {
```

Style nit.

(I'm not sure why `cargo fmt` doesn't catch these! I know it doesn't do well with macros, so maybe the `named_params!` are tripping it up in the rest of the file? 🤔)